fix: roll out tests

eliobischof opened the pull request:
Depends on
- https://github.com/zitadel/zitadel/pull/6252
- https://github.com/zitadel/zitadel/pull/6242
Rerunning tests fails until https://github.com/zitadel/zitadel/pull/6237 is released

Closes #91
Contributes to #85

muhlemmer reviewed (changes requested):
This was a rather big pull request, so I hope I didn't miss anything.

The main comment is about the change from `string` to `interface{}` types in `RunLifecyleTest`  `checkRemoteProperty` function argument, but incorrect handling in the implementing functions. I hope it doesn't give much extra work and let me know if you need help.

eliobischof commented on .github/workflows/pull_request.yaml:
```diff
@@ -17,39 +17,42 @@ jobs:
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.19'
```
done

eliobischof commented on zitadel/v2/org/resource_test.go:
```diff
@@ -0,0 +1,60 @@
+	"fmt"
+	"testing"
+
+	"github.com/zitadel/zitadel-go/v2/pkg/client/zitadel/org"
+
+	"github.com/zitadel/zitadel-go/v2/pkg/client/zitadel/admin"
```
enforcing formatted imports now

eliobischof commented on zitadel/v2/action/resource_test.go:
```diff
@@ -0,0 +1,58 @@
+				return err
+			}
+			actual := remoteResource.GetAction().GetScript()
+			if actual != expect {
```
done

eliobischof commented on zitadel/v2/action/resource_test.go:
```diff
@@ -0,0 +1,58 @@
+			}
+			actual := remoteResource.GetAction().GetScript()
+			if actual != expect {
+				return fmt.Errorf("expected %s, but got %s", expect, actual)
```
The types are explicit, now, so this is ok now

eliobischof commented:
> The tests are still too flaky. I think we have to fix that first.

@muhlemmer sorry, I changed quite a bit since your last review.
Besides working in your review proposals, I also fixed the flakiness and improved idempotency for test runs.

Where we depend on eventually consistent read functions, we can't guarantee empty plans directly after apply. Also, the test suite either allows to expect an empty plan or a non-empty plan, but not a dynamic condition. I couldn't find a better solution than basically just ignore these scenarios where they could occur:
https://github.com/zitadel/terraform-provider-zitadel/blob/test-all-resources/zitadel/v2/helper/test_utils/lifecyletest.go#L84-L90

I observed these cases only on `zitadel_login_texts` and `zitadel_default_login_texts` and only on the GitHub runner, not locally. But all tests that are potentially flaky for this reason are marked successful when this exact error occurs.

eliobischof commented:
@muhlemmer all five test runs were successful 
![image](https://github.com/zitadel/terraform-provider-zitadel/assets/12727842/1699bcba-825d-46d6-a748-6e9c6b0e8dd9)
